pkg/controller/registry/reconciler/grpc.go

```diff
@@ -185,15 +186,12 @@ func (c *GrpcRegistryReconciler) ensureUpdatePod(source grpcCatalogSourceDecorat
 	if !source.Poll() {
 		return nil
 	}
-	var updateFlag bool
 	currentLivePods := c.currentPods(source)
 	currentUpdatePods := c.currentUpdatePods(source)
 
 	for _, updatePod := range currentUpdatePods {
 		if updatePodByDigest(updatePod, currentLivePods) {
```

Comment:
We need to rework this loop a bit due to a few potential problems here:
1. There is a possibility that there are multiple update pods that have different digest compared to live pod. You either need to break out (hence the original `break` statement) of the loop when you find one or you risk promoting multiple update pods into serving pods.
2. The line #205 may fail on the second iteration if there are multiple qualified update pods and you delete the living pod on the first iteration because the living pod is gone.

My suggestion here is NOT to delete or promote anything during the loop. Instead, keep track of which update pod is the current candidate and check for the creation timestamp to see which one is created the latest in case there are multiple candidate and update the current candidate to latest candidate as the loop goes on. After the loop is done, promoting the current candidate to living pod, remove it from the update pods slice, deleting previous living pod and then delete all remaining update pods in the slice.

Reply:
Hey, so I reworked the PR immediately after I pushed -- sorry about that, I didn't think you would review so fast...

@ecordell said he was comfortable with a synchronous update loop that checks if the image has changed and immediately deletes the old catalog. this makes the logic much easier. see the 2nd commit for the rework.

Reply:
I thought without some sort of waiting time, the pod digest is often empty as it does take a bit of time for crio to pull the image down. The index image is unfortunately not as small as bundle due to included binaries. We did run into this issue back then during testing due to the issue I just described. Maybe you can run this test on an actual cluster to verify.

Reply:
Yes -- in the latest commit I basically put a blocking loop that waits for the container to be have an `imageID` that we are comparing. My takeaway from our meeting was that this was ok given that we want to have 2 catalog source pods on the cluster for a minimal amount of time and are OK with waiting.

I do have a `TODO` to bail out somehow if that update pod never gets scheduled and running successfully though

Reply:
Maybe @ecordell can clarify but what I got from Evan was we can't wait around and block the queue. So now I'm not sure if I misunderstood what he said or not.

Reply:
I think conceptually if we are syncing a catalog source and we see that its ready for an update, holding up the reconciliation loop for a small amount of time to see if an update is actually ready to roll out is a fair tradeoff. The user wants their updated catalog to be in use as soon as possible. It does effect other catalogs in the loop but I think its ok

Reply:
Looks like we should requeue until the pod is ready per evan's comment -- will update